akp/types/types.go

```diff
@@ -483,6 +499,20 @@ func toClusterDataAPIModel(ctx context.Context, diagnostics *diag.Diagnostics, c
 			}
 		}
 	}
+
+	directClusterSpec := &v1alpha1.DirectClusterSpec{}
```

Comment:
Will this have an empty string `""` as the type and then lead to the invalid argument error? The logic below only checks DirectClusterSpec if it's not nil; and if it's nil, an empty struct with type "" will be sent to the API.